### PR TITLE
Add bip32 serializations to account

### PR DESCRIPTION
**Pull request by mtps**
* Add bip32 serialization to account
* Add publicKeyOnly, testnet, and curve parameters where appropriate in Wallet functions.

### CODE REVIEW

File: hdwallet/src/main/kotlin/io/provenance/hdwallet/wallet/Wallet.kt [hunk 1]
```diff
@@ -6,35 +6,40 @@ import io.provenance.hdwallet.bip39.DeterministicSeed
...
 import io.provenance.hdwallet.bip44.PathElements
+import io.provenance.hdwallet.common.hashing.sha256
+import io.provenance.hdwallet.ec.CURVE
+import io.provenance.hdwallet.ec.Curve
 import io.provenance.hdwallet.ec.ECKeyPair
 import io.provenance.hdwallet.encoding.base58.base58DecodeChecked
 import java.security.KeyException
 
+/**
+ * Wallets are the root key representation used to derive [Account]s.
```
Comment (mwoods-figure) [inline hdwallet/src/main/kotlin/io/provenance/hdwallet/wallet/Wallet.kt:17]: 👍  I really like the combination of KDoc and dokka for documentation. If devops adopts Nexus pro, you can view the HTML of javadoc jars directly from within Nexus. Imagine being able to link to the Javadoc from the repo README!

File: hdwallet/src/main/kotlin/io/provenance/hdwallet/wallet/Wallet.kt [hunk 2]
```diff
@@ -79,31 +92,72 @@ interface Wallet {
...
             mnemonicWords = mnemonicWords,
-            testnet = testnet
+            publicKeyOnly = publicKeyOnly,
+            testnet = testnet,
+            curve = curve,
         )
     }
 }
 
+/**
```
Comment (mwoods-figure) [inline hdwallet/src/main/kotlin/io/provenance/hdwallet/wallet/Wallet.kt:117]: Thanks for this documentation. Super super helpful

**Review by mwoods-figure — APPROVED** [review]

File: hdwallet/src/main/kotlin/io/provenance/hdwallet/wallet/Wallet.kt (hunk 2 above)
Comment (mtps) [reply hdwallet/src/main/kotlin/io/provenance/hdwallet/wallet/Wallet.kt:117]: 👍

File: hdwallet/src/main/kotlin/io/provenance/hdwallet/wallet/Wallet.kt (hunk 1 above)
Comment (mtps) [reply hdwallet/src/main/kotlin/io/provenance/hdwallet/wallet/Wallet.kt:17]: That would be really useful! This one goes up to maven central though, not nexus.I haven't dug too much into their javadoc jar support yet.

**Pull request merged**